distribution/std-lib/Base/src/Data/Number/Extensions.enso

```diff
@@ -102,3 +103,16 @@ Decimal.parse : Text -> Decimal | Nothing
 Decimal.parse text =
     Panic.recover (Double.parseDouble [text]) . catch (_ -> Nothing)
 
+## Parses a textual representation of an integer into an integer number.
+   Returns `Nothing` if the text does not represent a valid integer.
+Integer.parse : Text -> Integer | Nothing
+Integer.parse text =
+    Panic.recover (Java_Integer.parseInt [text]) . catch (_ -> Nothing)
+
+## Parses a textual representation of a number into an integer or decimal number.
+   Returns `Nothing` if the text does not represent a valid number.
+Number.parse : Text -> Number | Nothing
+Number.parse text =
+    case Integer.parse text of
+        Nothing -> Decimal.parse text
+        value   -> value
```

Comment:
These, and `Decimal.parse` should use the dataflow errors mechanism, and return an `Unparseable` (or similar) error.